### PR TITLE
Updates to match `cedar` PR#260

### PR DESCRIPTION
**Pull request by khieta**
*Issue #, if available:*

*Description of changes:*

Updates to match [cedar#260](https://github.com/cedar-policy/cedar/pull/260). That PR updates the Rust `is_authorized` interface to distinguish between parsing vs. authorization errors, so now the Java `isAuthorized` function will throw a `BadRequestException` on a parse error rather than returning a Deny decision w/ parse errors hidden in the diagnostics.

Build failures are expected since the corresponding `cedar` PR is not yet merged.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CedarJavaFFI/src/interface.rs [hunk 1]
```diff
@@ -173,7 +173,7 @@ mod test {
             },
             "principal": null,
             "action": "Action::\"view\"",
-            "resource": "Resource:\"thing\""
+            "resource": "Resource::\"thing\""
```
Comment (khieta) [inline CedarJavaFFI/src/interface.rs:176]: Interesting case caught by this change. This test was returning Deny due to a parse error.

File: CedarJava/src/test/java/com/cedarpolicy/SharedIntegrationTests.java [hunk 2]
```diff
@@ -353,20 +354,29 @@ private void executeJsonQueryTest(
...
+        } catch (BadRequestException e) {
+            // In case of a parse error, isAuthorized and validate will throw `BadRequestException`.
+            // A few of our corpus tests currently fall into this category because a function call
+            // (e.g., `foo()`) is considered a parse error when the function name is unrecognized.
+
+            // TODO: this current fails for 48 test cases
+            // Some of these failures are expected (due to the comment above) but the "invalid" token ones
+            // are a little odd -- it looks like some policies being sent to the authorizer aren't the same
+            // as what's in the corpus files. Some issue with unicode?
+            throw new AssertionError("parse error: " + e.getMessage());
```
Comment (khieta) [inline CedarJava/src/test/java/com/cedarpolicy/SharedIntegrationTests.java:375]: I plan to address this TODO before merging... but looking for suggestions for the best approach. What can we assert in the case where there is a parse error?

**Comment by khieta** [comment]
Update: I think this PR is ready for review, but I'll leave it as a draft until we finalize the versioning for the Java code (since this will have to be the next major version).

**Comment by khieta** [comment]
Marking as ready-for-review now that we've started a `release/2.3.x` branch

**Review by cdisselkoen — APPROVED** [review]

**Comment by khieta** [comment]
Update: CI passes now that the correspnding change has been merged into `cedar`. Waiting on one final approval to merge.

**Review by andrewmwells-amazon — APPROVED** [review]

**Pull request merged**